pkg/materialize/database.go

```diff
@@ -71,6 +72,9 @@ func DatabaseId(conn *sqlx.DB, obj MaterializeObject) (string, error) {
 }
 
 func ScanDatabase(conn *sqlx.DB, id string) (DatabaseParams, error) {
+	// Get the database id from the state id
+	id, _ = utils.ExtractId(id)
```

Comment:
Nit: But I would remove the `ExtractId` from within the Scan functions. I feel like the `materialize` package should try to be separate from the Terraform functionality and combining them makes it harder to follow. We may also want to think of naming our variables around `id` since there is now the id within Materialize and the Terraform id.

Reply:
Agreed. Better to have the `id` here be a Materialize database ID, IMO, rather than a TF provider database ID.

Reply:
@benesch @dehume just made all of the requested changes. Let me know what you think. If this is ok, I will start with all of the rest of the resources.

Reply:
> just made all of the requested changes. Let me know what you think.

Looks great to me!

> If this is ok, I will start with all of the rest of the resources.

Godspeed.